File: CS-Examples/01_Quick guide/CreateAnExcelWithFiveSheets.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace CreateAnExcelWithFiveSheets
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {

            // Record the current time as the starting point
            DateTime start = DateTime.Now;

            // Create a new workbook
            Workbook workbook = new Workbook();

            // Create five empty sheets in the workbook
            workbook.CreateEmptySheets(5);

            // Iterate over each sheet in the workbook
            for (int i = 0; i < 5; i++)
            {
                // Get the current sheet
                Worksheet sheet = workbook.Worksheets[i];

                // Set the name of the sheet using the index
                sheet.Name = "Sheet" + i.ToString();

                // Populate the sheet with data in a grid-like pattern
                for (int row = 1; row <= 150; row++)
                {
                    for (int col = 1; col <= 50; col++)
                    {
                        // Set the text in each cell of the sheet using the row and column numbers
                        sheet.Range[row, col].Text = "row" + row.ToString() + " col" + col.ToString();
                    }
                }
            }

            // Specify the file name for saving the workbook
            String result = "CreateAnExcelWithFiveSheets_result.xlsx";

            // Save the workbook to a file with Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object
            workbook.Dispose();

            // Record the current time as the ending point
            DateTime end = DateTime.Now;

            // Calculate the time taken to create the file and display it as a message box
            TimeSpan time = end - start;
            MessageBox.Show("File has been created successfully! \n" + "Time consumed (Seconds): " + time.TotalSeconds.ToString());
            
            // View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/01_Quick guide/CreateAnExcelWithFiveSheets.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace CreateAnExcelWithFiveSheets
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Record the start time
            Dim start As Date = Date.Now

            ' Create a new Excel workbook object.
            Dim workbook As New Workbook()
            workbook.CreateEmptySheets(5)

            For i As Integer = 0 To 4
                ' Get the worksheet at the current index
                Dim sheet As Worksheet = workbook.Worksheets(i)
                ' Set the worksheet name 
                sheet.Name = "Sheet" & i.ToString()

                ' Fill in data by iterating through rows and columns
                For row As Integer = 1 To 150
                    For col As Integer = 1 To 50
                        sheet.Range(row, col).Text = "row" & row.ToString() & " col" & col.ToString() ' Set text content in the cell
                    Next col
                Next row
            Next i

            ' Specify the file name for the result
            Dim result As String = "CreateAnExcelWithFiveSheets_result.xlsx"

            ' Save the workbook to a file
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Record the end time
            Dim [end] As Date = Date.Now

            ' Calculate the elapsed time
            Dim time As TimeSpan = [end].Subtract(start)

            MessageBox.Show("File has been created successfully! " & vbLf & "Time consumed (Seconds): " & time.TotalSeconds.ToString())

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
